src/microns_to_nwb/tools/stimulus/stimulus.py

```diff
@@ -0,0 +1,28 @@
+import numpy as np
+from hdmf.backends.hdf5 import H5DataIO
+from phase3 import nda
+from pynwb.image import OpticalSeries
+
+
+def add_stimulus(scan_key, nwb):
+    # The stimulus images were synchronized with field 1 frame times
+    timestamps = (nda.FrameTimes() & scan_key).fetch1("frame_times")
+    field_of_view = (nda.Field() & scan_key).fetch("um_height", "um_width", as_dict=True)[0]
+    width_in_meters = field_of_view["um_width"] / 1e6
+    height_in_meters = field_of_view["um_height"] / 1e6
+    movie = (nda.Stimulus & scan_key).fetch1("movie")
+    # The movie dimensions are (height, width, number of frames), for NWB it should be
+    # transposed to (number of frames, width, height)
+    movie_transposed = movie.transpose(2, 1, 0)
+    optical_series = OpticalSeries(
+        name="visual stimulus",
+        description="The visual stimulus is composed of natural movies ~30 fps in grayscale format.",
+        distance=np.nan,  # unknown
+        field_of_view=[width_in_meters, height_in_meters],
```

Comment:
There are showing up as `NaN` in the example file - was this a new addition since you made that last file?

Reply:
These should be `NaN`, from discussion. Database does not know.

Reply:
This should be resolved now, I updated the file at https://gui-staging.dandiarchive.org/dandiset/206031/draft/